insalan/tournament/models.py

```diff
@@ -637,6 +652,12 @@ def save(self,*args,**kwargs):
         super().save(*args,**kwargs)
         self.team.refresh_validation()
 
+    def delete(self,*args,**kwargs):
+        if self.team.captain == self:
+            self.team.captain = None
```

Comment:
A team cannot not have a captain. That is a violation of the invariant of a Team, and will result in an unmodifiable team.

Reply:
This is handled by the team save method (https://github.com/InsaLan/backend-insalan.fr/blob/af80d9b38903541f409e61b8d91302c825ed01a2/insalan/tournament/models.py#L554C14-L554C18).

I tried something during my tests like having a default value and on delete : use default but it was clashing with django admin panel because you have to create a team with no player at first.

One thing we can add is deleting a team in the view (not in the model because of the same admin panel) if the last player call the delete endpoint.

Reply:
Empty teams are now deleted when last member leave it with 4916c1400000f15d8826d185178f00d5a61635f7